core/src/wsv.rs

```diff
@@ -98,7 +101,8 @@ pub struct WorldStateView {
     pub metric_tx_amounts: std::cell::Cell<f64>,
     /// Count of how many mints, transfers and burns have happened.
     pub metric_tx_amounts_counter: std::cell::Cell<u64>,
-
+    /// Reference to the pending tx queue from sumeragi.
+    queue: Arc<Queue>,
```

Comment:
if queue is now part of wsv then you should remove queue from `sumeragi` and have it access queue through wsv

Reply:
Should I do the same for kura then, as it's also present in both?